packages/blocks/countdown/src/app.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useEffect, useState, useRef } from "react";
+
+import { BlockComponent } from "blockprotocol/react";
+import DatePicker from "react-datepicker";
+import {
+  isFuture,
+  isPast,
+  formatDistanceToNow,
+  formatDistanceToNowStrict,
+  formatRelative,
+  format,
+} from "date-fns";
+
+import "react-datepicker/dist/react-datepicker.css";
+
+type Range = {
+  start: Date | null;
+  end: Date | null;
+};
+
+type AppProps = {
+  interval: {
+    start?: Date | string;
+    end?: Date | string;
+  };
+  selectsRange: boolean;
+  strict: boolean;
+};
+
+// const Checkbox = ({ label, value, onChange }) => {
+//   return (
+//     <label>
+//       <input type="checkbox" checked={value} onChange={onChange} />
+//       {label}
+//     </label>
+//   );
+// };
+
+const formatDate = (date: Date, rangeSelection: boolean) => {
+  if (rangeSelection) {
+    return format(date, "PPP");
+  } else {
+    return formatRelative(date, new Date());
+  }
+};
+
+const formatRange = (range: Range) => {
+  if (range.start !== null && range.end !== null) {
+    return `${formatDate(range.start, range.end !== null)} - ${formatDate(
+      range.end,
+      range.end !== null,
+    )}`;
+  } else {
+    return "";
+  }
+};
+
+const calculateTime = (range: Range, strict: boolean) => {
+  if (range.start === null) {
+    return {
+      prefix: "",
+      offset: "Please select a date",
+      postfix: "",
+    };
+  }
+
+  let relative;
+  let date;
+  if (isFuture(range.start)) {
+    date = range.start;
+    relative = "until";
+  } else if (range.end !== null && isPast(range.end)) {
+    date = range.end;
+    relative = "since";
+  } else if (range.end === null) {
+    date = range.start;
+    relative = "since";
+  } else {
+    return {
+      prefix: "",
+      offset: formatRange(range),
+      postfix: "is now",
+    };
+  }
+
+  let distance;
+  if (strict) {
+    distance = formatDistanceToNowStrict(date);
+  } else {
+    distance = formatDistanceToNow(date, { includeSeconds: true });
+  }
+
+  return {
+    prefix: `${distance} ${relative}`,
+    offset: formatDate(date, range.end !== null),
+    postfix: "",
+  };
+};
+
+export const App: BlockComponent<AppProps> = ({
+  interval = {
+    start: null,
+    end: null,
+  },
+  selectsRange = false,
+  strict = false,
+  entityId,
+  accountId,
+  updateEntities,
+}) => {
+  const [range, setRange] = useState<Range>({
+    start: !interval.start ? null : new Date(interval.start),
+    end: !interval.end ? null : new Date(interval.end),
+  });
+  const [rangeSelection, setRangeSelection] = useState(selectsRange);
+
+  const [clock, setClock] = useState({ now: new Date() });
+  const tick = () => setClock({ now: new Date() });
+
+  useEffect(() => {
+    setRange({
+      start: !interval.start ? null : new Date(interval.start),
+      end: !interval.end ? null : new Date(interval.end),
+    });
+    setRangeSelection(selectsRange);
+  }, [interval, selectsRange, strict]);
```

Comment:
I updated the update logic in 43bc756141b57b26aff3197a48727cbce5768d26. It's required to have this snippet when having multiple tabs of the same workspace open. If one entity got updated, other pages will refresh as well.

Reply:
`useState` shouldn't update the value if it doesn't change. However, `Interval === Interval` always returns `false` somehow. I made it separated types and it's now working as expected.

Reply:
The reason for this is that your `Interval` is an object, and object equality in JS is checking that they point to the same address in memory (i.e. the same object by identity, not contents).

To be honest comparing `interval.start === localInterval.start` is not reliable either, since `Date` is also an object, and every time you call `new Date(something)` you are creating an object, i.e. 

```javascript
new Date("2020") === new Date("2020") // false, these are not equal, they are different objects in memory
```

The way to compare objects in JS (without a library) is to convert them to a primitive representation, e.g. a string or number, and compare those. In the case of a `Date` you can use the method `getTime` or `valueOf` (they are equivalent) to get the number of ms since the UNIX epoch, and compare those.

```javascript
new Date("2020").getTime() === new Date("2020").getTime() // true, these numbers are the same
```

None of this matters since you changed the implementation, but thought I would explain.

Reply:
Thanks for the explaination!